### PR TITLE
convert test from request type to controller type

### PR DESCRIPTION
**Pull request by danielnottingham**
# Related Issue
- convert tests from request type to controller type #15

# Additional Info
- any additional information or context

# Checklist 
- [x] Unit Tests
- [x] Test coverage screenshot
- [ ] Deploy link:
Screenshot from 2020-09-16 09-56-52

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -3,13 +3,80 @@
...
+    user_attributes = FactoryBot.attributes_for(:user)
+    it 'create a new user' do
+      post :create, params: user_attributes
+      expect(response).to redirect_to(action: :index)
+    end
+
+    it 'not create a new user' do
+      user_attributes = { name: 'Sara' }
+      post :create, params: user_attributes
+      expect(response).to render_template(:new)
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:24]: check that user.errors really contain the errors

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -3,13 +3,80 @@
...
+  end
+
+  context 'GET #edit' do
+    let!(:user) { create(:user) }
+
+    it 'should success render edit' do
+      get :edit, params: { id: user.id }
+      expect(response).to render_template(:edit)
+      expect(assigns(:user)).to be_a(User)
+    end
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:35]: add case when edit fails and check that user.errors really contain the errors

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -3,13 +3,80 @@
...
+    let!(:user) { create(:user) }
+
+    it 'should update user info' do
+      params = { name: 'Edimo' }
+
+      put :update, params: { id: user.id, name: 'Edimo'}
+      user.reload
+      expect(user.name).to eq(params[:name])
+      expect(response).to redirect_to(action: :index)
+    end
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:48]: add case when edit fails and check that user.errors really contain the errors

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -3,13 +3,80 @@
...
+      expect(assigns(:users)).to_not be_empty
+    end
+  end
+
+  context 'DELETE #destroy' do
+    let!(:user) { create(:user) }
+
+    it 'should delete user' do
+      delete :destroy, params: { id: user.id }
+      expect(response).to redirect_to(action: :index)
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:75]: check that user doesnt exist anymore

**Comment by edimossilva** [comment]
deploy link and coverage screenshot not working

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:24]: still missing this

File: spec/controllers/users_controller_spec.rb [hunk 5]
```diff
@@ -3,13 +3,89 @@
 RSpec.describe UsersController, type: :controller do
 
   context 'GET #new' do
-    it 'should success' do
+    it 'should success render new' do
       get :new
       expect(response).to have_http_status(200)
+      expect(assigns(:user)).to be_a(User)
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:9]: remove this

Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:9]: also check the redirect

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -3,13 +3,89 @@
...
+      expect(assigns(:user)).to be_a_new(User)
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:10]: `be_a_new`, please explain why to have this ?

**Review by edimossilva — CHANGES_REQUESTED** [review]
Please resolve all conversations

File: spec/controllers/users_controller_spec.rb (hunk 6 above)
Comment (danielnottingham) [reply spec/controllers/users_controller_spec.rb:10]: it was another way of testing, different from the line above

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:23]: hey you need to check that user.errors really contain the errors

**Review by edimossilva — CHANGES_REQUESTED** [review]
@danielnottingham, you need to check that user.errors really contain the errors, or are they not present because a new user replace the current user and the errors become empty?

**Comment by danielnottingham** [comment]
> @danielnottingham, you need to check that user.errors really contain the errors, or are they not present because a new user replace the current user and the errors become empty?

@edimossilva 
Sorry but I still don't understand. Should I check the user.errors for the GET NEW context?

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:23]: something similar to 
```
    expect(assigns(:user).errors.full_messages).to eq(["Name can't be blank", 'Name is invalid'])
```

**Comment by edimossilva** [comment]
> > @danielnottingham, you need to check that user.errors really contain the errors, or are they not present because a new user replace the current user and the errors become empty?
> 
> @edimossilva
> Sorry but I still don't understand. Should I check the user.errors for the GET NEW context?

I added a comment to it, please check if you understand now @danielnottingham

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:24]: @danielnottingham resolve this!!!!!!

**Review by edimossilva — APPROVED** [review]

**Pull request merged**